client/Encounter/Encounter.ts

```diff
@@ -244,6 +232,12 @@ export class Encounter {
 
     public Save: (name: string) => SavedEncounter<SavedCombatant> = (name: string) => {
         let activeCombatant = this.ActiveCombatant();
+        this.durationTags
+            .filter(t => !t.Visible())
+            .forEach(t => {
+                t.Remove();
+                this.durationTags.splice(this.durationTags.indexOf(t), 1);
+            });
```

Comment:
This isn't quite right. Save shouldn't be side-affecting, it should only strip the tags off of the SavedEncounter that it returns. Since the encounter is saved after every action this winds up immediately trimming off the expired tags.

Reply:
ooooh, didn't realize this was happening automatically on every action. It seems like it might just be reasonable then to do something like this below:

```
return {
    ...
    Combatants: this.Combatants().map<SavedCombatant>(c => {
        ...
        Tags: c.Tags().filter(t => t.Visible()).map(t => ({
           ...
```

Is that a proper approach?

Reply:
Yep, that's how I'd do it. Verify the correct behavior after making the change, you should see it work.